lib/money/money.rb

```diff
@@ -84,7 +84,9 @@ def -(other)
   end
 
   def *(numeric)
-    raise TypeError, "#{numeric} is not Numeric" unless numeric.is_a?(Numeric)
+    unless numeric.is_a?(Numeric)
+      Money.deprecate("Multiplying Money with #{numeric.class.name} is deprecated and will be removed in the next major release.")
```

Comment:
`unless numeric.is_a?(Numeric)` So if it's not a numeric we show the deprecation warning